src/modal-dialog.cpp

```diff
@@ -48,7 +49,7 @@ void Modal_Dialog::initialize() {
 	// Initialize dialog
 	_dialog->box(OS_BG_BOX);
 	_dialog->resizable(NULL);
-	_dialog->callback((Fl_Callback *)close_cb, this);
+	_dialog->callback((Fl_Callback *)cancel_cb, this);

```

Comment:
This resulted in a bug on Linux (and Mac) where pressing the X button in the modal window's title bar would act like pressing OK rather than like pressing Cancel.

The only reason this bug doesn't happen on Windows is because modal windows don't have an X button in their title bar.